### PR TITLE
LPS-189893 Publishing apps on startup

### PR DESCRIPTION
**Pull request by 4lejandrito**
https://liferay.atlassian.net/browse/LPS-189893.

**How to test**
- Create an API Builder App and publish it.
- Restart the server.
- The app should be published automatically.

I don't know how to test this automatically.

Rebased on top of https://github.com/brianchandotcom/liferay-portal/pull/137461 and https://github.com/brianchandotcom/liferay-portal/pull/137486. Blocked those are merged

### CODE REVIEW

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/instance/lifecycle/APIApplicationPublisherPortalInstanceLifecycleListener.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * @author Alejandro Tardín
+ */
+@Component(service = PortalInstanceLifecycleListener.class)
+public class APIApplicationPublisherPortalInstanceLifecycleListener
+	extends BasePortalInstanceLifecycleListener {
+
+	@Override
+	public void portalInstanceRegistered(Company company) throws Exception {
+		List<ObjectEntry> apiApplicationObjectEntries =
+			_objectEntryHelper.getObjectEntries(
```
[the remark this answers is not in the document]
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/instance/lifecycle/APIApplicationPublisherPortalInstanceLifecycleListener.java:41]: I think it makes sense. I'll do it.

[the remark this answers is not in the document]
Comment (4lejandrito) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/instance/lifecycle/APIApplicationPublisherPortalInstanceLifecycleListener.java]: Done by @ccorreagg! Thanks!

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Review by ccorreagg — APPROVED** [review]
LGTM!

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
Rebased on top of https://github.com/brianchandotcom/liferay-portal/pull/137755.

**Comment by 4lejandrito** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/137802

**Pull request closed without merging**